### PR TITLE
[DO-7442] Add gossip cluster implementation for Digital Ocean cloud provider

### PR DESCRIPTION
**Pull request by srikiz**
As part of this PR, it supports multiple HA masters for DO by adding gossip seed support. The necessary changes are already done on the etcd-manager side - https://github.com/kopeio/etcd-manager/pull/254.
I have tested this locally with 3 master nodes, and was able to see the seeding happening Ok. 
This PR doesn't address adding a load balancer for the HA masters. That will be coming up next.

### CODE REVIEW

File: protokube/pkg/gossip/do/seeds.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import (
+	"fmt"
+	//"errors"
```
Comment (bittopaz) [inline protokube/pkg/gossip/do/seeds.go]: Maybe it's better to remove this commented lines in import.

Comment (srikiz) [reply protokube/pkg/gossip/do/seeds.go]: Sure, updated.

File: pkg/model/components/etcdmanager/model.go [hunk 2]
```diff
@@ -181,6 +181,7 @@ func parseManifest(data []byte) ([]runtime.Object, error) {
 }
 
 // Until we introduce the bundle, we hard-code the manifest
+
```
Comment (zetaab) [inline pkg/model/components/etcdmanager/model.go]: remove this change, its not needed

File: pkg/resources/digitalocean/resources.go [hunk 3]
```diff
@@ -195,7 +196,8 @@ func listDNS(cloud fi.Cloud, clusterName string) ([]*resources.Resource, error)
 	}
 
 	if domainName == "" {
-		return nil, fmt.Errorf("failed to find domain for cluster: %s", clusterName)
+		klog.Info("Domain Name is empty. Possible that the cluster is configured as gossip..")
```
Comment (zetaab) [inline pkg/resources/digitalocean/resources.go]: we could check here does the `clusterName` endswith k8s.local. Then we can say is it gossip cluster or not. If not then we should return real error

File: protokube/pkg/gossip/do/seeds.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+				klog.V(2).Infof("Tag matched for droplet tag =%s. Getting private IP", p.tag)
+				ip, err := droplet.PrivateIPv4()
+				if err == nil {
+					klog.V(2).Infof("Appending a seed for cluster tag:%s, with ip=%s", p.tag, ip)
+					seeds = append(seeds, ip)
+				} else {
+					klog.V(2).Infof("Ah ...Private IP failed for tag=%s, error=%v", p.tag, err)
+				}
+			} else {
+				klog.V(2).Infof("Tag NOT matched for droplet tag =%s. and pTag=%s", dropTag, p.tag)
```
Comment (zetaab) [inline protokube/pkg/gossip/do/seeds.go]: maybe loglevel 4?

File: protokube/pkg/protokube/do_volume.go [hunk 5]
```diff
@@ -259,6 +314,13 @@ func getMetadataDropletID() (string, error) {
 	return getMetadata(dropletIDMetadataURL)
 }
 
+func getMetadataDropletTags() ([]string, error) {
+
+	tagString, error := getMetadata(dropletIDMetadataTags)
```
Comment (zetaab) [inline protokube/pkg/protokube/do_volume.go]: Use `err` instead of `error` just in case

File: protokube/pkg/protokube/do_volume.go [hunk 6]
```diff
@@ -259,6 +314,13 @@ func getMetadataDropletID() (string, error) {
...
+	dropletTags := strings.Split(tagString, "\n")
+	return dropletTags, error
```
Comment (zetaab) [inline protokube/pkg/protokube/do_volume.go]: you could combine these two rows as `return strings.Split(tagString, "\n"), err`

File: pkg/model/components/etcdmanager/model.go (hunk 2 above)
Comment (srikiz) [reply pkg/model/components/etcdmanager/model.go]: Thanks, updated.

**Comment by justinsb** [comment]
Thanks @srikiz ... this is great stuff!

/approve
/lgtm

**Pull request merged**